src/scope.ts

```diff
@@ -671,12 +683,10 @@ export class FunctionScope extends ClosureEnvironment {
     /* ori func name iff func is declare */
     oriFuncName: string | undefined = undefined;
     debugLocations: SourceMapLoc[] = [];
-    // iff this FunctionScope is specialized
-    private _genericOwner?: FunctionScope;
 
-    constructor(parent: Scope) {
+    constructor(parent: Scope | null) {
```

Comment:
Why would it be possible to have a null parent scope?

Reply:
done.
This does seem redundant and wrong